core/gui/src/app/workspace/component/code-editor-dialog/code-editor.component.ts

```diff
@@ -169,11 +183,153 @@ export class CodeEditorComponent implements AfterViewInit, SafeStyle, OnDestroy
+          if (!model || !selection) {
+            return;
+          }
+          // All the code in Python UDF
+          const allcode = model.getValue();
+          // Content of user selected code
+          const code = model.getValueInRange(selection);
+          // Start line of the selected code
+          const lineNumber = selection.startLineNumber;
+          this.handleTypeAnnotation(code, selection, ed as monaco.editor.IStandaloneCodeEditor, lineNumber, allcode);
+        },
+      });
+    }
     if (this.language == "python") {
       this.connectLanguageServer();
     }
   }
 
+  private handleTypeAnnotation(
+    code: string,
+    range: monaco.Range,
+    editor: monaco.editor.IStandaloneCodeEditor,
+    lineNumber: number,
+    allcode: string
+  ): void {
+    this.aiAssistantService
+      .getTypeAnnotations(code, lineNumber, allcode)
+      .pipe(takeUntil(this.workflowVersionStreamSubject))
+      .subscribe({
+        next: (response: TypeAnnotationResponse) => {
+          const choices = response.choices || [];
+          if (choices.length > 0 && choices[0].message && choices[0].message.content) {
+            this.currentSuggestion = choices[0].message.content.trim();
+
+            let acceptButton: HTMLButtonElement | null = null;
+            let declineButton: HTMLButtonElement | null = null;
+
+            this.currentCode = code;
+            this.currentRange = range;
+            this.showAnnotationSuggestion = true;
+
+            setTimeout(() => {
+              const position = editor.getScrolledVisiblePosition(range.getStartPosition());
+              const popupElement = document.querySelector(".annotation-suggestion") as HTMLElement;
+
+              // Make sure the UI pop up next to the selected argument
+              if (popupElement && position) {
+                popupElement.style.top = `${position.top + 100}px`;
+                popupElement.style.left = `${position.left + 100}px`;
+              }
+
+              const cleanup = () => {
+                if (acceptButton) acceptButton.removeEventListener("click", acceptListener);
+                if (declineButton) declineButton.removeEventListener("click", declineListener);
+                this.showAnnotationSuggestion = false;
+              };
+
+              const acceptListener = () => {
+                this.acceptCurrentAnnotation();
+                cleanup();
+              };
+
+              const declineListener = () => {
+                cleanup();
+              };
+
+              acceptButton = document.querySelector(".accept-button") as HTMLButtonElement;
+              declineButton = document.querySelector(".decline-button") as HTMLButtonElement;
+
+              if (acceptButton && declineButton) {
+                acceptButton.addEventListener("click", acceptListener, { once: true });
+                declineButton.addEventListener("click", declineListener, { once: true });
+              }
```

Comment:
major: we are using angular. please write it as an angular component, it will be much easier to define a button and its handling functions in such a pop-up component. avoid using some raw HTML event bindings.